metadata-ingestion/src/datahub/ingestion/source/thrift.py

```diff
@@ -0,0 +1,141 @@
+from dataclasses import dataclass, field
+from typing import Generator, Iterable, Union
+
+from antlr4 import CommonTokenStream, InputStream
+
+from datahub.configuration.common import ConfigModel
+from datahub.emitter.mcp import MetadataChangeProposalWrapper
+from datahub.ingestion.api.source import Source, SourceReport
+from datahub.ingestion.api.workunit import MetadataWorkUnit, UsageStatsWorkUnit
+from datahub.ingestion.source.dist.ThriftLexer import ThriftLexer  # type: ignore
+from datahub.ingestion.source.dist.ThriftParser import ThriftParser  # type: ignore
+from datahub.metadata.com.linkedin.pegasus2avro.events.metadata import ChangeType
+from datahub.metadata.com.linkedin.pegasus2avro.schema import (
+    NumberType,
+    OtherSchema,
+    SchemaField,
+    SchemaFieldDataType,
+    SchemaMetadata,
+    StringType,
+)
+
+
+class ThriftSourceConfig(ConfigModel):
```

Comment:
Is there anything else that we want to make configurable?

Reply:
So far, no

Reply:
TODO

Reply:
TODO: add a directory option, potentially a git repo in the future.